source/dub/platform.d

```diff
@@ -102,6 +103,10 @@ enum string archCheck = q{
 	version(LoongArch64) ret ~= "loongarch64";
 	version(LoongArch_SoftFloat) ret ~= "loongarch_softfloat";
 	version(LoongArch_HardFloat) ret ~= "loongarch_hardfloat";
+	version(WebAssembly) {
+		version(D_LP64) ret ~= "wasm64";
+		else ret ~= "wasm32";
```

Comment:
making everything non-64-bit be wasm32 doesn't seem right, should this maybe instead be:
```
version (WebAssembly) ret ~= "webassembly";
version (WebAssembly) version (D_LP64) ret ~= "wasm64";
```

so we are also closer to the D Version identifiers and make it easier for us to add new wasm variants in the future?

Reply:
There is only wasm32 and wasm64 in terms of architecture; and they are sufficiently different that they need to be seperated. (Otherwise the linker will be upset)

Reply:
oh I see, also read some more here and I think this should be fine